ui/lib/core/addon/components/modal.js

```diff
@@ -10,12 +10,32 @@
  * @param {function} onClose - onClose is the action taken when someone clicks the modal background or close button (if shown).
  * @param {string} [title] - This text shows up in the header section of the modal.
  * @param {boolean} [showCloseButton=false] - controls whether the close button in the top right corner shows.
+ * @param {String} type=null  - The header type. This comes from the message-types helper.
```

Comment:
I'm not sure it matters, but should this be a lower case "string" instead of "String".  Also, it looks like maybe there's an extra space between "null  -"